### PR TITLE
Introduce option to display shorter author data on EDS item detail page

### PR DESCRIPTION
**Pull request by alexklbuckley**
- Add DetailPageFormat config to EDS.ini
- Add short author display to EDS core.phtml files
- Follow-up to PR3841

Sponsored-By: Auckland University of Technology, New Zealand

### CODE REVIEW

File: themes/bootstrap3/templates/RecordDriver/EDS/core.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordDriver/EDS/core.phtml]: I don't think $edsConfig is defined in the context of this template, unless I am overlooking something.

Also, the .ini file says the default is Short, but the default here is Long. We should pick one and be consistent.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @alexklbuckley! I haven't tried hands-on testing yet because I suspect it's not quite working yet based on my comments below. I'll take a closer look once you have addressed this review.

File: themes/bootstrap3/templates/RecordDriver/EDS/core.phtml
Comment (alexklbuckley) [reply themes/bootstrap3/templates/RecordDriver/EDS/core.phtml]: thanks @demiankatz ! sorry about the delay.

I've pushed a follow-up addressing both of your points here.

**Comment by berndoberknapp** [comment]
I would suggest to not show the AuInfo item when DetailPageFormat = 'Short' is used. When the AuInfo item is present it contains author affiliation information referenced from the Au item.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @alexklbuckley! I'm marking this back to "request changes" until @berndoberknapp's suggestion is addressed, and then I'll take a closer look myself. :-)

**Comment by alexklbuckley** [comment]
Hi @demiankatz and @berndoberknapp , I have been doing some local testing of this enhancement and have pushed a follow-up fixing a few bugs that I noticed.

Now when DisplayPageFormat is set to:
- 'Short' - Then 'BibRecord/BibRelationships/HasContributorRelationships/*PersonEntity/Name/NameFull' data is displayed. Just like  for https://github.com/vufind-org/vufind/pull/3841
- 'Long' - Then the current author data that is displayed on upstream dev branch is displayed.

Can you please confirm for me Bernd if this is the functionality fix you were wanting in https://github.com/vufind-org/vufind/pull/3881#issuecomment-2325271586 ?

**Review by demiankatz — APPROVED** [review]
@alexklbuckley, I pushed up a few minor adjustments:

1.) I combined some of the comments in EDS.ini to reduce redundancy
2.) I combined some of the CSS in EDS.css to reduce redundancy
3.) I added missing EDS.css changes to the bootstrap5 theme
4.) I tweaked some whitespace in the core.phtml template

I'm approving this because it seems to work correctly for me, but I'll wait for @berndoberknapp's approval before actually merging in case there are any further details I have overlooked.

**Comment by alexklbuckley** [comment]
> @alexklbuckley, I pushed up a few minor adjustments:
> 
> 1.) I combined some of the comments in EDS.ini to reduce redundancy 2.) I combined some of the CSS in EDS.css to reduce redundancy 3.) I added missing EDS.css changes to the bootstrap5 theme 4.) I tweaked some whitespace in the core.phtml template
> 
> I'm approving this because it seems to work correctly for me, but I'll wait for @berndoberknapp's approval before actually merging in case there are any further details I have overlooked.

Thanks @demiankatz !

**Comment by berndoberknapp** [comment]
I won't be able to test the changes before next week, but my comment regaring AuInfo doesn't seem to be addressed yet?

**Comment by demiankatz** [comment]
> I won't be able to test the changes before next week, but my comment regaring AuInfo doesn't seem to be addressed yet?

I'm not very familiar with the API responses. Is AuInfo a different `$item['group']` value that we might encounter? Is the idea that if we're set to short mode, this information will be irrelevant and should be bypassed? If that's the case, could you perhaps share a couple of example record IDs that include this element?

**Comment by berndoberknapp** [comment]
> I'm not very familiar with the API responses. Is AuInfo a different `$item['group']` value that we might encounter? Is the idea that if we're set to short mode, this information will be irrelevant and should be bypassed? If that's the case, could you perhaps share a couple of example record IDs that include this element?

Yes, AuInfo is a separate group with additional information about the authors. While looking for examples I noticed that there are different cases. For many EBSCOhost databases like Academic Search or Business Source Au includes references to AuInfo which only contains "Author Affiliations", and in that case AuInfo doesn't make much sense when Short is used (e.g. aph,178692351). But there are databases like Psyndex where AuInfo contains information like "Author ID", "Author Email" or "Country of First Author" and where the author name is repeated, so in that case it would make sense to display the information (e.g. pdx,0421957). I can try to check whether this is consistent across databases, but not before next week.

**Comment by demiankatz** [comment]
Thanks, @berndoberknapp! It sounds like it could get complicated to try to display AuInfo only for certain databases. It may be sufficient to just hide it in "short" mode as you propose -- we would just likely want to make note of this in the config settings.

**Comment by demiankatz** [comment]
I've just pushed up a change to hide the AuInfo group in short author display mode. I think this is probably okay since most people will use the defaults (short in search results, long on record view) and it makes the most sense in that context.... but of course I'm open to change if people have other ideas. @alexklbuckley and @berndoberknapp, let me know if you're comfortable merging things in this current form!

**Comment by alexklbuckley** [comment]
> I've just pushed up a change to hide the AuInfo group in short author display mode. I think this is probably okay since most people will use the defaults (short in search results, long on record view) and it makes the most sense in that context.... but of course I'm open to change if people have other ideas. @alexklbuckley and @berndoberknapp, let me know if you're comfortable merging things in this current form!

Thank you very much @demiankatz !

Yes, from my perspective, I am happy for this to be merged in it's current form :)

**Comment by berndoberknapp** [comment]
> I've just pushed up a change to hide the AuInfo group in short author display mode. I think this is probably okay since most people will use the defaults (short in search results, long on record view) and it makes the most sense in that context.... but of course I'm open to change if people have other ideas. @alexklbuckley and @berndoberknapp, let me know if you're comfortable merging things in this current form!

That seems to be the most sensible solution, I'm of course fine with merging it.

**Comment by demiankatz** [comment]
Thanks, @alexklbuckley and @berndoberknapp -- merging now!

**Comment by alexklbuckley** [comment]
Thanks @demiankatz and @berndoberknapp !

**Pull request merged**
